pkg/node/acl.go

```diff
@@ -0,0 +1,106 @@
+/*
+ *
+ * Copyright © 2022 Dell Inc. or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package node
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+	"regexp"
+	"strings"
+
+	"github.com/dell/gopowerstore"
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+var (
+	execCommand = executeCommand
+)
+
+func validateAndSetACLs(ctx context.Context, nasName string, client gopowerstore.Client, acls string, dir string) (bool, error) {
+	aclsConfigured := false
+	if nfsv4ACLs(acls) {
+		if nfsv4NasServer(ctx, client, nasName) {
+			if err := setNfsv4Acls(acls, dir); err != nil {
+				log.Error(fmt.Sprintf("can't assign NFSv4 ACLs to folder %s: %s", dir, err.Error()))
+				return false, err
+			}
+			aclsConfigured = true
+		} else {
+			return false, status.Errorf(codes.Internal, "can't assign NFSv4 ACLs to folder %s: NAS server is not NFSv4 enabled", dir)
+		}
+	} else {
+		return false, status.Errorf(codes.Internal, "can't assign ACLs to folder %s: invalid NFSv4 ACL format %s", dir, acls)
+	}
+
+	return aclsConfigured, nil
+}
+
+func posixMode(acls string) bool {
+	if matched, _ := regexp.Match(`\d{3,4}`, []byte(acls)); matched {
+		return true
+	}
+	return false
+}
+
+func nfsv4ACLs(acls string) bool {
+	aclsList := strings.Split(acls, ",")
+	for _, acl := range aclsList {
+		matched, err := regexp.Match(`([ADUL]:\w*:[\w.]*[@]*[\w.]*:\w*)`, []byte(acl))
+		if !matched || err != nil {
+			return false
+		}
+	}
+	return true
+}
+
+func setNfsv4Acls(acls string, dir string) error {
+	command := []string{"nfs4_setfacl", "-s", acls, dir}
+	log.Infof("NFSv4 ACL command: %s \n", strings.Join(command, " "))
+	outStr, err := execCommand(command)
+	log.Infof("NFSv4 ACL output: %s \n", string(outStr))
+	return err
+}
+
+func executeCommand(command []string) ([]byte, error) {
```

Comment:
exec.Command should be used individually wherever necessary since the command parameters needs to be validated to prevent command injections.

Reply:
1. In this file, exec.Command is used only once. 
2. The arguments are validated before calling it. 
3. The wrapper is used here for the sake of mocking during unit test execution. 
Please let me know if it still needs to be changed.